File: LR_1-2/4.pas
program ReshiMne;

var a,b:integer; var res:real;

begin
  // Принимаем значения
  writeln('Введите a и b');
  readln(a, b);
  
  if (a = 0) and (b = 0) then
  begin
    println('INF')
  end;
  
  if (a = 0) and (b <> 0) then
  begin
    println('NO')
  end;
  
  if (a <> 0) and (b <> 0) then
  begin
    res := (-b)/a;
    println(res);
  end;
end.

File: LR/LR_1-2/4.py
a = int(input("Введите a"))
b = int(input("Введите b"))

if a == 0 and b == 0:
    print(f'INF')

if a == 0 and b != 0:
    print(f'NO')

if a != 0 and b != 0:
    res = -b / a
    print(res)